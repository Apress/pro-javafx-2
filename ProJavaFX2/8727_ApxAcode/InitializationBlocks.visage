/*
 * @test
 * @run
 */

var counter = 0;

class Base {
  var i: Integer;
  init {
    println("Step {counter}: Base init block.  i = {i}.");
    counter++;
  }
  postinit {
    println("Step {counter}: Base postinit block.  i = {i}.");
    counter++;
  }
}

class Derived extends Base {
  var str: String;
  init {
    println("Step {counter}: Derived init block.  i = {i}, str = {str}.");
    counter++;
  }
  postinit {
    println("Step {counter}: Derived postinit block.  i = {i}, str = {str}.");
    counter++;
  }
}

var o = Derived {
  i: {
    println("Step {counter}: i initialized to 1024.");
    counter++;
    1024;
  }
  str: {
    println('Step {counter}: str initialized to "Hello, World".');
    counter++;
    "Hello, World";
  }
}
