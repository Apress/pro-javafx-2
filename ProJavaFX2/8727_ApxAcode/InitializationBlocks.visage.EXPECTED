Step 0: i initialized to 1024.
Step 1: str initialized to "Hello, World".
Step 2: Base init block.  i = 1024.
Step 3: Derived init block.  i = 1024, str = Hello, World.
Step 4: Base postinit block.  i = 1024.
Step 5: Derived postinit block.  i = 1024, str = Hello, World.
